server/src/main/java/org/elasticsearch/common/geo/CentroidCalculator.java

```diff
@@ -32,12 +45,13 @@
     private double sumY;
     private int count;
 
-    public CentroidCalculator() {
+    public CentroidCalculator(Geometry geometry) {
```

Comment:
I think we cannot do that because we need to support arrays of shapes and therefore the calculation of the centroid needs the calculator. This is something it should change in #49887 so I think we should not worry too much. Depending in how we decide to calculate the centroid we would need to change this class accordingly.

Reply:
I agree that if we are to change how the calculation is structured, it is best to do that in the dedicated calculator issue Ignacio mentioned